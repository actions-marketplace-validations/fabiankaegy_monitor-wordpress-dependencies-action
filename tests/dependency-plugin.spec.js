const DependencyPlugin = require('../lib/dependency-plugin.js');

test('DependencyPlugin', async () => {
	const plugin = new DependencyPlugin({
		pattern: '**/*.asset.php'
	});

	const cwd = process.cwd();

	const newSizes = await plugin.readFromDisk(cwd);

	const oldSizes = await plugin.readFromDisk(cwd);

	const diff = await plugin.getDiff(oldSizes, newSizes);

	expect(diff).toMatchSnapshot();	
});

test('DependencyPluginOutput', async () => {
	const plugin = new DependencyPlugin({
		pattern: '**/*.asset.php'
	});

	const newSizes = {
        'sample/index.asset.php': [ 'wp-block-editor', 'wp-blocks', 'wp-element', 'wp-primitives' ],
      }

	const oldSizes = {
        'sample/index.asset.php': [ 'wp-block-editor', 'wp-blocks', 'wp-components', 'wp-element' ],
      }

	const diff = await plugin.getDiff(oldSizes, newSizes);

	expect(diff).toMatchSnapshot();
});
