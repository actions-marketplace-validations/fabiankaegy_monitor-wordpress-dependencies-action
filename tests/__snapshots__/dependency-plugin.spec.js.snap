// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependencyPlugin 1`] = `
[
  {
    "dependency": "wp-block-editor, wp-blocks, wp-components, wp-element",
    "difference": "",
    "filename": "fixtures/after/index.asset.php",
  },
  {
    "dependency": "wp-block-editor, wp-blocks, wp-components",
    "difference": "",
    "filename": "fixtures/before/index.asset.php",
  },
]
`;
